mkcodes.py

```diff
@@ -79,43 +113,55 @@ def get_files(inputs):
         elif path.suffix in markdown_extensions:
             yield path, path.parent
 
-def add_inits_to_dir(path):
+
+def add_inits_along_path(from_path, to_path):
     """Recursively add __init__.py files to a directory
-    This compensates for https://bugs.python.org/issue23882 and https://bugs.python.org/issue35617
+    This compensates for https://bugs.python.org/issue23882
+    and https://bugs.python.org/issue35617
     """
-    for child in path.rglob('*'):
-        if child.is_dir():
-            (child / '__init__.py').touch()
-
+    to_path = to_path.expanduser().resolve()
+    from_path = from_path.expanduser().resolve()
+    (to_path / '__init__.py').touch()
+    if to_path == from_path:
+        return
+    if to_path.relative_to(from_path):
+        if from_path != to_path.parent:
+            add_inits_along_path(from_path, to_path.parent)
```

Comment:
It seems like this function could be simplified to:

```py
def add_inits_along_path(from_path, to_path):
    to_path = to_path.expanduser().resolve()
    from_path = from_path.expanduser().resolve()
    (to_path / '__init__.py').touch()
    if to_path != from_path:
        add_inits_along_path(from_path, to_path.parent)
```

Any objections?

Reply:
I think relative_to is a cheap and fast safety check - the implementation is based on the string representation of the path. As the code is right now, it's very safe, but I do like some safety checks when recursing UP that I'm less concerned about in recursing down.

See my above code for slightly simpler, but avoids __init__.py in the output base directory and has a safety check.

Reply:
this is not a *strong* objection - if you like that version, it's fine and will work great.